SSL HandShake failed

SSL Handshake failed every time since restricting only to TLSv1.2 on server part configuration

```
javax.net.ssl.SSLHandshakeException:` Handshake failed
02-19 14:00:21.152 W/System.err:     at com.android.org.conscrypt.OpenSSLSocketImpl.startHandshake(OpenSSLSocketImpl.java:444)
02-19 14:00:21.152 W/System.err:     at org.apache.http.conn.ssl.SSLSocketFactory.createSocket(SSLSocketFactory.java:406)
02-19 14:00:21.152 W/System.err:     at org.apache.http.impl.conn.DefaultClientConnectionOperator.openConnection(DefaultClientConnectionOperator.java:219)
02-19 14:00:21.153 W/System.err:     at org.apache.http.impl.conn.AbstractPoolEntry.open(AbstractPoolEntry.java:172)
02-19 14:00:21.153 W/System.err:     at org.apache.http.impl.conn.AbstractPooledConnAdapter.open(AbstractPooledConnAdapter.java:130)
02-19 14:00:21.153 W/System.err:     at org.apache.http.impl.client.DefaultRequestDirector.executeOriginal(DefaultRequestDirector.java:1334)
02-19 14:00:21.153 W/System.err:     at org.apache.http.impl.client.DefaultRequestDirector.execute(DefaultRequestDirector.java:700)
02-19 14:00:21.153 W/System.err:     at org.apache.http.impl.client.AbstractHttpClient.execute(AbstractHttpClient.java:560)
02-19 14:00:21.153 W/System.err:     at org.apache.http.impl.client.AbstractHttpClient.execute(AbstractHttpClient.java:492)
02-19 14:00:21.153 W/System.err:     at org.apache.http.impl.client.AbstractHttpClient.execute(AbstractHttpClient.java:470)
02-19 14:00:21.154 W/System.err:     at ekylibre.APICaller.Instance.getJSONArray(Instance.java:267)
02-19 14:00:21.154 W/System.err:     at ekylibre.APICaller.Instance.getJSONArray(Instance.java:238)
02-19 14:00:21.154 W/System.err:     at ekylibre.APICaller.Intervention.all(Intervention.java:56)
02-19 14:00:21.154 W/System.err:     at ekylibre.APICaller.SyncAdapter.pullIntervention(SyncAdapter.java:454)
02-19 14:00:21.154 W/System.err:     at ekylibre.APICaller.SyncAdapter.onPerformSync(SyncAdapter.java:138)
02-19 14:00:21.154 W/System.err:     at android.content.AbstractThreadedSyncAdapter$SyncThread.run(AbstractThreadedSyncAdapter.java:272)
02-19 14:00:21.155 W/System.err: Caused by: javax.net.ssl.SSLProtocolException: SSL handshake terminated: ssl=0xdda74dc0: Failure in SSL library, usually a protocol error
02-19 14:00:21.155 W/System.err: error:10000410:SSL routines:OPENSSL_internal:SSLV3_ALERT_HANDSHAKE_FAILURE (external/boringssl/src/ssl/s3_pkt.c:641 0xdda21360:0x00000001)
02-19 14:00:21.155 W/System.err: error:1000009a:SSL routines:OPENSSL_internal:HANDSHAKE_FAILURE_ON_CLIENT_HELLO (external/boringssl/src/ssl/s3_clnt.c:800 0xdbcb9a9f:0x00000000)
02-19 14:00:21.155 W/System.err:     at com.android.org.conscrypt.NativeCrypto.SSL_do_handshake(Native Method)
02-19 14:00:21.155 W/System.err:     at com.android.org.conscrypt.OpenSSLSocketImpl.startHandshake(OpenSSLSocketImpl.java:364)
02-19 14:00:21.155 W/System.err: 	... 15 more
```

--- a/app/src/main/java/ekylibre/zero/AuthenticatorActivity.java
+++ b/app/src/main/java/ekylibre/zero/AuthenticatorActivity.java
@@ -11,6 +11,8 @@ import android.view.View;
 import android.widget.EditText;
 import android.widget.Toast;
 
+import com.google.android.gms.security.ProviderInstaller;
+
 import ekylibre.APICaller.Token;
 import ekylibre.exceptions.UnauthorizedException;
 import ekylibre.zero.home.MainActivity;
@@ -22,7 +24,7 @@ import ekylibre.zero.home.MainActivity;
  *
  * It sends back to the Authenticator the result.
  */
-public class AuthenticatorActivity extends AccountAuthenticatorActivity {
+public class AuthenticatorActivity extends AccountAuthenticatorActivity implements ProviderInstaller.ProviderInstallListener {
 
     public final static String KEY_ACCOUNT_PASSWORD    = "accountPassword";
     public final static String KEY_AUTH_TOKEN_TYPE     = "authTokenType";
@@ -49,6 +51,8 @@ public class AuthenticatorActivity extends AccountAuthenticatorActivity {
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.login);
+
+        ProviderInstaller.installIfNeededAsync(this, this);
 
         mAccountNameEdit     = (EditText) findViewById(R.id.accountName);
         mAccountPasswordEdit = (EditText) findViewById(R.id.accountPassword);
@@ -156,5 +160,15 @@ public class AuthenticatorActivity extends AccountAuthenticatorActivity {
             startActivity(mainIntent);
         }
         finish();
+    }
+
+    @Override
+    public void onProviderInstalled() {
+        if (BuildConfig.DEBUG) Log.d("zero", TAG + "Security provider updated");
+    }
+
+    @Override
+    public void onProviderInstallFailed(int i, Intent intent) {
+        if (BuildConfig.DEBUG) Log.d("zero", TAG + "Cannot update security provider");
     }
 }

